src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1809,10 +1816,16 @@ public static bool TryLoadLibrary(string libraryPath, out IntPtr handle)
         /// <exception cref="System.BadImageFormatException">Thrown if the library is not valid.</exception>        
         public static IntPtr LoadLibrary(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
         {
-            RuntimeAssembly runtimeAssembly = (assembly != null) ? ((RuntimeAssembly)assembly).GetNativeHandle() : null;
-            bool throwOnError = true;
-            uint searchPathFlag =  searchPath.HasValue ? (uint)searchPath.Value : 0;
-            return LoadLibraryByName(libraryName, runtimeAssembly, searchPath.HasValue, searchPathFlag, throwOnError);
+            if (libraryName == null)
+                throw new ArgumentNullException(nameof(libraryName));
+            if (assembly == null)
+                throw new ArgumentNullException(nameof(assembly));
+            
+            return LoadLibraryByName(libraryName, 
+                                     ((RuntimeAssembly)assembly).GetNativeHandle(), 
```

Comment:
This may rather be:

```
            if (!(assembly is RuntimeAssembly runtimeAssembly))
                throw new ArgumentException(SR.Argument_MustBeRuntimeAssembly);
```

Reply:
(or something similar that actually compiles :-)